node-cpp.md

```diff
@@ -1,66 +1,115 @@
+
+```shell
+UV_THREADPOOL_SIZE=1 node index.js
+```
+
+And the output is
+
+```
+royal with cheese
+royal with cheese
+royal with cheese
+royal with cheese
+```
 
 ### How is it different than a C++ project?
 
 A Node.js addon is still a Node module. Users still interact with it as if they are writing Javascript (i.e. `var awesome = require('awesome')`), but the library will tend to pass much of the logic into C++ workers, which are highly performant, then return information back into a javascript interface. Bottom line, the user of your library never has to write or interact with C++.
 
+### Native Abstractions for Node.js (NAN)
+
+To swing between Node and C++, the Node community maintains a project called [_NAN_](https://github.com/nodejs/nan) (Native Abstractions for Node.js) that simplifies running different versions of Node and, subsequently, v8. NAN is a header-only C++ library that provides a set of Macros for developing Node.js addons. Check out the [usage](https://github.com/nodejs/nan#usage) guidelines.
+
+More examples of how to port C++ libraries to node can be found at [nodejs.org/api/addons](http://nodejs.org/api/addons.html). See https://nodesource.com/blog/c-add-ons-for-nodejs-v4/ for a detailed summary of the origins of Nan. And see http://blog.reverberate.org/2016/10/17/native-extensions-memory-management-part2-javascript-v8.html for a good introduction to v8 that is not specific to node.js.
+
 ### Examples
 
 All of the following libraries are installable in a Node.js environment, but execute much of their logic in C++:
 
 * [Node Mapnik](https://github.com/mapnik/node-mapnik) - creating map tiles
 * [Node OSRM](https://github.com/Project-OSRM/node-osrm) - directions & routing
 * [sqlite](https://github.com/mapbox/node-sqlite3) - asynchronous, non-blocking SQLite3 bindings
-* [vtinfo](https://github.com/mapbox/vtinfo) - reads and returns general information about a vector tile buffer
 * [Node GDAL](https://github.com/naturalatlas/node-gdal) - geographic operations
+* [vtquery](https://github.com/mapbox/vtquery) - query vector tiles
+* [vtinfo](https://github.com/mapbox/vtinfo) - reads and returns general information about a vector tile buffer
 
 ### Developing addons
 
-Developing an addon requires Node.js, NPM, and a C++ compiler.
-
-* makefile: home to all of the development commands for building binaries, installing dependencies, and running tests
+Developing an addon requires Node.js, NPM, and a C++ compiler. Check out node-cpp-skel's "extended tour" for an up-to-date and opinionated approach to developing an addon. Additinally, the repository has docs on [running benchmarks](https://github.com/mapbox/node-cpp-skel/blob/master/docs/benchmarking.md), [publishing binaries](https://github.com/mapbox/node-cpp-skel/blob/master/docs/publishing-binaries.md), and a breakdown of all the [necessary components](https://github.com/mapbox/node-cpp-skel/blob/master/docs/extended-tour.md#configuration-files) to make it run.
```

Comment:
Add link to [node-cpp-skel's "extended tour"](https://github.com/mapbox/node-cpp-skel/blob/master/docs/extended-tour.md)